src/auth/ainizeAuth.ts

```diff
@@ -0,0 +1,67 @@
+import Ain from '@ainblockchain/ain-js';
+import Ainize from '@ainize-team/ainize-js';
+
+export default class AinizeAuth {
+  private static instance: AinizeAuth;
+  private ain: Ain | null = null;
+  private ainize: Ainize | null = null;
+  private loggedIn: boolean = false;
+
+  private constructor() {}
+
+  static getInstance() {
+    if (!this.instance) {
+      this.instance = new AinizeAuth();
+    }
+    return this.instance;
+  }
+
+  init(ain: Ain, ainize: Ainize) {
```

Comment:
(suggestion) how about defining it as static so that users can initialize it before instantiating?

Reply:
I think initializing with `init()` function, apart from the constructor, is not a good practice..
So I refactored `AinizeAuth` class to inject ain/ainize objects at method level for cleaner and better(?).

Reply:
Ah, static functions cannot access member variables. In that approach, maybe the members variables can also be defined static.. 
I just see now login/logout functions accepts ain and/or ainize objects as parameters. Just looks fine, thanks.

Reply:
Thank you for quick reviews!